Packs/AzureResourceGraph/Integrations/AzureResourceGraph/AzureResourceGraph.py

```diff
@@ -0,0 +1,281 @@
+    Args:
+        response: The response from the API.
+        limit: Maximum number of objects to retrieve.
+        page: Page number
+    Returns:
+        Return a list of objects from the response according to the page and limit per page.
+    """
+    if page_size > MAX_PAGE_SIZE:
+        page_size = MAX_PAGE_SIZE
+    
+    starting_index = (page_number - 1) * page_size
+    ending_index = starting_index + page_size
+    return response[starting_index:ending_index]
+
+
+def validate_connection_params(tenant: str = None, auth_and_token_url:str = None, is_self_deployed: bool = False, enc_key: str = None, certificate_thumbprint: str = None, private_key: str = None, subscription_id: str = None) -> None:
+    if not tenant or not auth_and_token_url:
+        raise DemistoException('Token and ID must be provided.')
+    
+    if not is_self_deployed and not enc_key:
+        raise DemistoException('Key must be provided. For further information see '
+                   'https://xsoar.pan.dev/docs/reference/articles/microsoft-integrations---authentication')
+    elif not enc_key and not (certificate_thumbprint and private_key):
+        raise DemistoException('Key or Certificate Thumbprint and Private Key must be providedFor further information see '
+                               'https://xsoar.pan.dev/docs/reference/articles/microsoft-integrations---authentication')
+    if not subscription_id:
+       raise DemistoException('A subscription ID must be provided.')
+
+
+def main():
+    params: dict = demisto.params()
+    args = demisto.args()
+    server = params.get('host', 'https://management.azure.com').rstrip('/')
+    tenant = params.get('cred_token', {}).get('password') or params.get('tenant_id')
+    auth_and_token_url = params.get('cred_auth_id', {}).get('password') or params.get('auth_id')
+    enc_key = params.get('cred_enc_key', {}).get('password') or params.get('enc_key')
+    certificate_thumbprint = params.get('cred_certificate_thumbprint', {}).get(
+        'password') or params.get('certificate_thumbprint')
+    private_key = params.get('private_key')
+    verify = not params.get('unsecure', False)
+    subscription_id = args.get('subscription_id') or params.get(
+        'cred_subscription_id', {}).get('password') or params.get('subscription_id')
+    proxy: bool = params.get('proxy', False)
+    self_deployed: bool = params.get('self_deployed', False)
+
+    validate_connection_params(tenant, auth_and_token_url, self_deployed, enc_key,
+                               certificate_thumbprint, private_key, subscription_id)
+    
+    ok_codes = (200, 201, 202, 204)
+
+    commands_without_args: Dict[Any, Any] = {
+        'test-module': test_module
+    }
+
+    commands_with_args: Dict[Any, Any] = {
+        'azure-rg-query': query_resources_command,
+        'azure-rg-list-operations': list_operations_command
+    }
+
+    commands_with_args_and_params: Dict[Any, Any] = {
+    }
```

Comment:
```suggestion
```

Removing as we usually do not pass the params but saves in the client what might be needed.